app/controllers/assignments_controller.rb

```diff
@@ -181,39 +180,43 @@ def update_github_permissions(assignment)
       access_token = session[:github_token]
       client = Octokit::Client.new(access_token: access_token)
       org_name = "AutograderFrontend"
-
-      # All assignments that need to be updated
       @users = User.all
 
-
       @users.each do |user|
         repo_identifier = "#{org_name}/#{assignment.repository_name}"
-
-        # Fetch the user's permission for the current assignment
         permission = Permission.find_by(user: user, assignment: assignment)
 
+        update_collaborator_permissions(client, repo_identifier, user, permission, assignment)
+      end
+    end
+
+    private
 
-        if permission.role == "no_permission" || permission.role.nil?
-          begin
-            client.remove_collaborator(repo_identifier, user.name)
-            Rails.logger.info "Removed collaborator #{user.name} from #{repo_identifier} due to 'none' permission"
-          rescue Octokit::Error => e
-            Rails.logger.error "Failed to remove collaborator #{user.name} from #{repo_identifier}: #{e.message}"
-            raise
-          end
-        elsif permission.role == "read_write" || permission.role == "read"
-          # Use ternary operator for selecting permission
-          github_permission = (permission.role == "read_write") ? "push" : "pull"
-          begin
-            client.add_collaborator(repo_identifier, user.name, permission: github_permission)
-            Rails.logger.info "Updated collaborator #{user.name} on #{repo_identifier} with #{github_permission} access"
-          rescue Octokit::Error => e
-            Rails.logger.error "Failed to update collaborator #{user.name} on #{repo_identifier}: #{e.message}"
-            raise
-          end
-        else
-          Rails.logger.error "Unknown permission role: #{permission.role} for user #{user.name} on assignment #{assignment.id}"
-        end
+    def update_collaborator_permissions(client, repo_identifier, user, permission, assignment)
```

Comment:
This function shows a flag for exceeding the number of arguments on code climate. Now, the `assignment` variable is used only for error logging.

Is there a possibility to avoid using `assignment` in the function argument?

Reply:
Yes, I will get rid of that.